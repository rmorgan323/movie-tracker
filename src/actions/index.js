import { getRecentMovies } from '../helper/getRecentMovies/getRecentMovies';
import addNewUser from '../helper/addNewUser/addNewUser';
import getUser from '../helper/getUser/getUser';
import checkUser from '../helper/checkUser/checkUser';
import getUserFavorites from '../helper/getUserFavorites/getUserFavorites';
import postNewFavoriteData from '../helper/postNewFavoriteData/postNewFavoriteData';
import deleteFavoriteData from '../helper/deleteFavoriteData/deleteFavoriteData';

export const makeMovieArray = movies => ({
  type: 'MAKE_MOVIE_ARRAY',
  movies
});

export const getMovies = () => async dispatch => {
  const newMovies = await getRecentMovies();
  dispatch(makeMovieArray(newMovies));
};

export const newUser = user => ({
  type: 'ADD_NEW_USER',
  user
});

export const setUser = user => ({
  type: 'SET_USER',
  user
});

export const fetchUsers = user => async dispatch => {
  const response = await addNewUser(user);
  if (response.status === 'success') {
    const user = await getUser(response.id);
    dispatch(setUser(user));
  } else {
    dispatch(signupFailure('error-signup'));
  }
};

export const signupFailure = msg => ({
  type: 'SIGNUP_ERROR',
  msg
});

export const checkForUser = user => async dispatch => {
  const response = await checkUser(user);
  if (response.status === 'success') {
    dispatch(setUser(response.data));
  } else {
    dispatch(signupFailure('error-login'));
  }
};

export const userLogout = () => ({
  type: 'USER_LOGOUT'
});

export const deletePost = (movie) => async dispatch => {
  const response = await deleteFavoriteData(movie.userId, movie.movieId);

  if (response.status === 'success') {
    dispatch(deleteFavorite(movie));
  }
};

export const deleteFavorite = (movie) => ({
  type: 'DELETE_FAVORITE',
  movie
});

export const addPost = (movie) => async dispatch => {
  const response = await postNewFavoriteData(movie);

  if (response.status === 'success') {
    dispatch(addFavorite(movie));
  }
};

export const addFavorite = (movie) => ({
  type: 'ADD_FAVORITE',
  movie
});

export const checkUserFavorites = (userId) => async dispatch => {
  const response = await getUserFavorites(userId);
  dispatch(setUserFavorites(response.data));
};

export const setUserFavorites = (favorites) => ({
  type: 'SET_USER_FAVORITES',
  favorites
});

