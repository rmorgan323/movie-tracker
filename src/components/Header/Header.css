.header {
  background-color: grey;
  height: 75px;
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  margin-bottom: 10px; }
  .header h1 {
    color: #fff; }
  .header .header {
    display: flex;
    flex-direction: row; }
  .header .login, .header .signup {
    position: absolute;
    right: 20px; }
  .header .login {
    top: 20px; }
  .header .signup {
    top: 40px; }

@media screen and (max-width: 400px) {
  .header {
    height: 100px; }
    .header h1 {
      width: 200px;
      text-align: center; } }
