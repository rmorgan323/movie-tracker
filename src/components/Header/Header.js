import React from 'react';
import { Link, NavLink } from 'react-router-dom';

const Header = () => {
    return (
        <div>
            <h1>Movie Tracker</h1>
            <NavLink to='/login'>Login</NavLink>
        </div>
    )
}

export default Header;