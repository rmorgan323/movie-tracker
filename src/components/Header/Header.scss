.header {
	background-color: grey;
	height: 75px;
	display: flex;
	align-items: center;
	justify-content: center;
	position: relative;
	margin-bottom: 10px;

	h1 {
		color: #fff;
	}

	.header {
		display: flex;
		flex-direction: row;
	}


	.login, .signup {
		position: absolute;
		right: 20px;
	}

	.login {
		top: 20px;		
	}

	.signup {
		top: 40px;
	}
}

@media screen and (max-width: 400px) {
	.header {
		height: 100px;

		h1 {
			width: 200px;
			text-align: center;
		}
	}
}