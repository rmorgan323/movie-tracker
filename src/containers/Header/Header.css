.header {
  background-color: rgba(255, 255, 255, 0.1);
  height: 75px;
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  margin-bottom: 10px; }
  .header h1 {
    color: #fff; }
  .header a {
    text-decoration: none; }
  .header button {
    position: absolute;
    left: 20px;
    top: 21px;
    height: 33px;
    border: none;
    border: none;
    cursor: pointer;
    font-weight: 700;
    color: #888;
    font-size: 15px;
    width: 120px;
    background-color: rgba(255, 255, 255, 0.7);
    transition: background-color .2s ease-in-out; }
  .header button:hover {
    background-color: #1fffff;
    transition: background-color .6s ease-in-out; }
  .header .login, .header .signup {
    position: absolute;
    right: 20px; }
  .header .login {
    top: 20px; }
  .header .signup {
    top: 40px; }
  .header .nav-link {
    text-decoration: none;
    color: white;
    text-transform: uppercase; }
  .header .nav-link:hover {
    color: #1fffff; }
  .header .login-message {
    color: #1fffff; }

@media screen and (max-width: 400px) {
  .header {
    height: 100px; }
    .header h1 {
      width: 200px;
      text-align: center; } }
