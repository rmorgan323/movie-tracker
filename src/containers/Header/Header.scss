.header {
	background-color: grey;
	height: 75px;
	display: flex;
	align-items: center;
	justify-content: center;
	position: relative;
	margin-bottom: 10px;

	h1 {
		color: #fff;
	}

	.login, .signup {
		position: absolute;
		right: 20px;
	}

	.login {
		top: 20px;
	}

	.signup {
		top: 40px;
	}

	.nav-link {
		text-decoration: none;
		color: white;
		text-transform: uppercase;
	}

	.nav-link:hover{
		color: #1fffff;
	}

	.login-message {
		color: #1fffff;
	}


}



@media screen and (max-width: 400px) {
	.header {
		height: 100px;

		h1 {
			width: 200px;
			text-align: center;
		}
	}
}