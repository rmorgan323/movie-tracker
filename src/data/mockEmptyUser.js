export const mockEmptyUser = {
  signedIn: false,
  userInfo: {
    id: '',
    name: '',
    password: '',
    email: '',
    favorites: []
  }
};
