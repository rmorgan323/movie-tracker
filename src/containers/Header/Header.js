import React, { Component } from 'react';
import { NavLink, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { userLogout } from '../../actions';
import { getMovies } from '../../actions';
import determineUser from '../../helper/determineUser/determineUser';
import './Header.css';

class Header extends Component {
	componentDidMount = () => {
    this.props.storeMovies()
  }

	noUserRender = () => {
		return (
			<div className="header-box">
	      <NavLink className="login nav-link" to='/login'>Login</NavLink>
	      <NavLink className="signup nav-link" to='/signup'>Sign-up</NavLink>
			</div>
		)
	}

	userRender = () => {
		return (
			<div className="header-box">
				<div className="login login-message">Welcome, {this.props.user.userInfo.name}</div>
				<NavLink to='/favorites'><button>FAVORITES</button></NavLink>
				<NavLink className="signup nav-link" to='/' onClick={this.props.logoutUser} >Sign-Out</NavLink>
			</div>
		)
	}

	render() {
		var headerLinks = determineUser(this.props.user)  ? this.userRender()
																											: this.noUserRender()
		return (
			<div className="header">
				<NavLink to='/' >
					<h1><span>MOVIE</span>TRACKER</h1>
				</NavLink>
				{headerLinks}
			</div>
		)
	}
}

export const mapStateToProps = store => {
  return {
    user: store.user
  };
};

export const mapDispatchToProps = (dispatch) => {
	return {
		logoutUser: () => dispatch(userLogout()),
		storeMovies: () => dispatch(getMovies())
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(Header);
