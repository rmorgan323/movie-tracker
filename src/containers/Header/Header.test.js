import { Header, mapStateToProps, mapDispatchToProps } from './Header';
import React from 'react';
import { shallow } from 'enzyme';
import { mockEmptyUser } from '../../data/mockEmptyUser';
import { mockRealUser } from '../../data/mockRealUser';
import * as actions from '../../actions';

global.localStorage = {
  getItem(keyword) {
    if (!global.localStorage[keyword]) {
      return null;
    }
    return JSON.stringify(global.localStorage[keyword]);
  },
  setItem(keyword, value) {
    global.localStorage[keyword] = value;
  }
};

describe('Header tests', () => {
  let renderedHeader;
  let mockStoreMovies;
  let mockLogout;

  beforeEach(() => {
    mockStoreMovies = jest.fn();
    mockLogout = jest.fn();

    renderedHeader = shallow(
      <Header
        user={mockEmptyUser}
        storeMovies={mockStoreMovies}
        logoutUser={mockLogout}
      />
    );
  });

  it('Should match the snapshot', () => {
    expect(renderedHeader).toMatchSnapshot();
  });

  it('Should have login and signup button with no user', () => {
    expect(renderedHeader.find('.login').length).toEqual(1);
    expect(renderedHeader.find('.signup').length).toEqual(1);
  });

  it('Should have a favorites button with user logged in', () => {
    expect(renderedHeader.find('button').length).toEqual(0);

    renderedHeader = shallow(
      <Header
        user={mockRealUser}
        storeMovies={mockStoreMovies}
        logoutUser={mockLogout}
      />
    );

    expect(renderedHeader.find('button').length).toEqual(1);
  });

  describe('mapStateToProps tests', () => {
    it('Should pull a user from store', () => {
      const mockStore = {
        user: mockRealUser
      };
      const result = mapStateToProps(mockStore);

      expect(result.user).toEqual(mockStore.user);
    });
  });

  describe('mapDispatchToProps tests', () => {
    let mockDispatch;
    let result;

    beforeEach(() => {
      mockDispatch = jest.fn();
      result = mapDispatchToProps(mockDispatch);
    });

    it('Should call dispatch when logoutUser is called', () => {
      result.logoutUser();

      expect(mockDispatch).toHaveBeenCalledWith(actions.userLogout());
    });

    it('Should call dispatch when storeMovies is called', async () => {
      const mockDispatch = jest.fn();
      const result = await mapDispatchToProps(mockDispatch);
      result.storeMovies();

      expect(mockDispatch).toHaveBeenCalled();
    });
  });
});
