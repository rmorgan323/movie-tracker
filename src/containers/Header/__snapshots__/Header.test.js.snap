// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    logoutUser={[Function]}
    storeMovies={[Function]}
    user={
        Object {
            "signedIn": false,
            "userInfo": Object {
              "email": "",
              "favorites": Array [],
              "id": "",
              "name": "",
              "password": "",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/"
>
          <h1>
                    <span>
                              MOVIE
                    </span>
                    TRACKER
          </h1>
</NavLink>,
        <div
          className="header-box"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="login nav-link"
                    to="/login"
          >
                    Login
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="signup nav-link"
                    to="/signup"
          >
                    Sign-up
          </NavLink>
</div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <h1>
            <span>
                        MOVIE
            </span>
            TRACKER
</h1>,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                MOVIE
</span>,
              "TRACKER",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MOVIE",
              },
              "ref": null,
              "rendered": "MOVIE",
              "type": "span",
            },
            "TRACKER",
          ],
          "type": "h1",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="login nav-link"
              to="/login"
>
              Login
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="signup nav-link"
              to="/signup"
>
              Sign-up
</NavLink>,
          ],
          "className": "header-box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Login",
              "className": "login nav-link",
              "to": "/login",
            },
            "ref": null,
            "rendered": "Login",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Sign-up",
              "className": "signup nav-link",
              "to": "/signup",
            },
            "ref": null,
            "rendered": "Sign-up",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/"
>
            <h1>
                        <span>
                                    MOVIE
                        </span>
                        TRACKER
            </h1>
</NavLink>,
          <div
            className="header-box"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="login nav-link"
                        to="/login"
            >
                        Login
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="signup nav-link"
                        to="/signup"
            >
                        Sign-up
            </NavLink>
</div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <h1>
              <span>
                            MOVIE
              </span>
              TRACKER
</h1>,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  MOVIE
</span>,
                "TRACKER",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "MOVIE",
                },
                "ref": null,
                "rendered": "MOVIE",
                "type": "span",
              },
              "TRACKER",
            ],
            "type": "h1",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="login nav-link"
                to="/login"
>
                Login
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="signup nav-link"
                to="/signup"
>
                Sign-up
</NavLink>,
            ],
            "className": "header-box",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Login",
                "className": "login nav-link",
                "to": "/login",
              },
              "ref": null,
              "rendered": "Login",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Sign-up",
                "className": "signup nav-link",
                "to": "/signup",
              },
              "ref": null,
              "rendered": "Sign-up",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
